app/src/main/kotlin/compiler/ast/StatementAst.kt

```diff
@@ -0,0 +1,34 @@
+package compiler.ast
+
+typealias BlockAst = List<StatementAst>
+
+sealed class StatementAst {
+    data class Evaluation(val expression: ExpressionAst) : StatementAst()
+
+    data class VariableDefinition(val variable: VariableAst) : StatementAst()
+    data class FunctionDefinition(val function: FunctionAst) : StatementAst()
+
+    data class Assignment(
+        val variableName: String,
+        val value: ExpressionAst
+    ) : StatementAst()
+
+    data class Block(val block: BlockAst) : StatementAst()
+
+    data class Conditional(
+        val condition: ExpressionAst,
+        val actionWhenTrue: BlockAst,
+        val actionWhenFalse: BlockAst?
+    ) : StatementAst()
+
+    data class Loop(
+        val condition: ExpressionAst,
+        val action: BlockAst
+    ) : StatementAst()
+
+    object LoopBreak : StatementAst()
+
+    object LoopContinuation : StatementAst()
+
+    data class FunctionReturn(val value: ExpressionAst?) : StatementAst()
```

Comment:
Shouldn't we always return some value (for `Unit` functions `zakończ` will be translated to `zwróć Unit`)?

Reply:
I think this translation should be performed by the type checker, and not by parser. The AST should reflect the code as seen by the programmer. For the same reason the field `returnType` in `FunctionAst` is optional.